File: BACK/src/Modules/Stakeholders/Explorer.Stakeholders.Core/Domain/ClubChallengeRequest.cs
using Explorer.BuildingBlocks.Core.Domain;
using Explorer.Stakeholders.Core.Domain.Enums;

namespace Explorer.Stakeholders.Core.Domain;

public class ClubChallengeRequest : Entity
{
    public Club? Challenger { get; init; }
    public long ChallengerId { get; init; }
    public Club? Challenged { get; init; }
    public long ChallengedId { get; init; }
    public ClubChallengeRequestStatus Status { get; init; }

    public ClubChallengeRequest(long challengerId, long challengedId, ClubChallengeRequestStatus status)
    {
        ChallengerId = challengerId;
        ChallengedId = challengedId;
        Status = status;
    }
}

File: FRONT/Explorer/src/app/feature-modules/tourist/model/club-challenge-request.ts
import { Club } from "./club.model";

export enum ClubChallengeStatus {
    PENDING = 'PENDING',
    ACCEPTED = 'ACCEPTED',
    DECLINED = 'DECLINED'
}

export interface ClubChallengeRequest {
    id?: number,
    challenger?: Club,
    challengerId: number,
    challenged?: Club,
    challengedId: number,
    status: ClubChallengeStatus
}